codec-http/src/main/java/io/netty/handler/codec/http/multipart/DefaultHttpDataFactory.java

```diff
@@ -261,7 +278,9 @@ public void cleanAllHttpData() {
         Iterator<Entry<HttpRequest, List<HttpData>>> i = requestFileDeleteMap.entrySet().iterator();
         while (i.hasNext()) {
             Entry<HttpRequest, List<HttpData>> e = i.next();
-            i.remove();
+
+            // Calling i.remove() here will cause "java.lang.IllegalStateException: Entry was removed"
+            // at e.getValue() below
 
             List<HttpData> fileToDelete = e.getValue();
             if (fileToDelete != null) {
```

Comment:
the reference count ownership of this class seems suspect. For example in `cleanRequestHttpData` and `cleanAllHttpData` we call `delete()`, but it seems like we should instead be calling `release()` (which then calls `delete()` if the reference count reaches 0).

Reply:
👍 Fixed and added test